File: Beecrowd/Beginner/1101 - 1190/1176 - Fibonacci Array/C++/1176_FibonacciArray.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int number;
    cin >> number;

    long long fibonacci[61];

    fibonacci[0] = 0;
    fibonacci[1] = 1;

    for (int i = 2; i <= 60; i++) {
        fibonacci[i] = fibonacci[i - 1] + fibonacci[i - 2];
    }

    while (number--) {
        int index;
        cin >> index;

        cout << "Fib(" << index << ") = " << fibonacci[index] << endl;
    }
    
    return 0;
}

File: Beecrowd/Beginner/1101 - 1190/1176 - Fibonacci Array/Python/1176_FibonacciArray.py
number = int(input())

fibonacci = [0] * 61
fibonacci[0] = 0
fibonacci[1] = 1

for i in range(2, 61):
    fibonacci[i] = fibonacci[i - 1] + fibonacci[i - 2]

for _ in range(number):
    index = int(input())
    print(f"Fib({index}) = {fibonacci[index]}")
